include/libtorrent/session_settings.hpp

```diff
@@ -1503,6 +1504,10 @@ namespace libtorrent
 		// the number of seconds a immutable/mutable item will be expired.
 		// default is 0, means never expires.
 		int item_lifetime;
+
+		// the number of seconds DHT routing table need to be refreshed, to
+		// maintain enough fresh nodes in routing table.
+		int refresh_timeout;
```

Comment:
now that I think about it. The original paper talks about refresh intervals in terms of how frequently a single node should be pinged, which iirc, was one hour. The 5 second counter here really is derived from 1 hour / number_of_nodes. It's an optimization to even out the cost of the routing table maintenance. I wonder if it would make more sense for the setting to be specified in the term of how often a single node is pinged.